tests/integration/suite/daprd/workflow/crossapp/orderingpreserved.go

```diff
@@ -0,0 +1,239 @@
+/*
+Copyright 2025 The Dapr Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package crossapp
+
+import (
+	"context"
+	"fmt"
+	"sync/atomic"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/dapr/dapr/tests/integration/framework"
+	"github.com/dapr/dapr/tests/integration/framework/process/workflow"
+	"github.com/dapr/dapr/tests/integration/suite"
+	"github.com/dapr/durabletask-go/api"
+	"github.com/dapr/durabletask-go/task"
+)
+
+func init() {
+	suite.Register(new(orderingpreserved))
+}
+
+// orderingpreserved demonstrates that activity execution order is preserved
+// across local and cross-app calls using an atomic slice to track order
```

Comment:
I'm not sure this test is covering much in terms of crossapp execution. The ordering is happening because of how the workflow code is structured, regardless of where the activities run. Or in other words, this is just how the workflows work, the `Await` is kind of enforcing the order.